code_generation/quantity.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class Quantity:
+    def __init__(self, name):
+        self.name = name
+        self.shifts = {}
+        self.children = {}
+        self.output_scopes = []
```

Comment:
I'd call it self.defined_for_scopes. 'output' does not make sense to me.

Reply:
renamed